### DISCUSSION TITLE
Offline Client __errors table

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#569] [Q&A]
I have a row in the __errors table in the offline sqlite database and I'm not sure what to do with it.

Anytime I try to sync, it does not hit the server, I just get the result which is in the __errors table.

What can be done with it ?

I've tried to use UpdateOperationAsync(), but this fails with

`            if (!await OperationsQueue.UpdateOperationAsync(error.Id, error.OperationVersion, item, cancellationToken).ConfigureAwait(false))
            {
                throw new InvalidOperationException("The operation has been updated and cannot be updated again.");
            }`

which is because there is no corresponding operation in the __operations table.

**Comment by adrianhall** [comment]
As mentioned in the other thread, I think we need some methods to deal with the errors table.  Any thoughts on this?

**Reply by henda79** [reply]
So how does the __errors table work ?

Is it if a Push fails, then it records the error for that table ? I came across this with the FK issues on the server side. It recorded a error 500 for the client and obviously there isn't much can be done on the client side.

I still think the code is wrong which I pasted in the other thread as it should match for the table been pushed only.

I'm not really sure what methods are needed as I'm not sure exactly how this all works. I've just started exploring the operations and errors table in a effort to try to create a decent sync resolution.

**Reply by adrianhall** [reply]
Let's switch over to discussing it in #571 so we can keep all the details together.
